llarp/util/threadpool.cpp

```diff
@@ -59,19 +59,25 @@ llarp_threadpool_wait(struct llarp_threadpool *pool)
 void
 llarp_threadpool_queue_job(struct llarp_threadpool *pool,
                            struct llarp_thread_job job)
+{
+  return llarp_threadpool_queue_job(pool, (std::bind(job.work, job.user)));
+}
+
+void
+llarp_threadpool_queue_job(struct llarp_threadpool *pool,
+                           std::function< void() > func)
 {
   if(pool->impl)
   {
-    while(!pool->impl->tryAddJob(std::bind(job.work, job.user)))
+    while(!pool->impl->tryAddJob(func))
     {
       std::this_thread::sleep_for(std::chrono::microseconds(1000));
     }
   }
   else
   {
     // single threaded mode
-    while(pool->jobs->tryPushBack(std::bind(job.work, job.user))
-          != llarp::thread::QueueReturn::Success)
+    while(pool->jobs->tryPushBack(func) != llarp::thread::QueueReturn::Success)
```

Comment:
not that this is a behaviour change, but if the queue is full, and this is single threaded, I think this could spin forever?

Reply:
I think that without the try it spins, but with the try it doesn't.  Either way if this spins forever I'd say the bigger problem is the threadpool size.

Reply:
also, in this specific case it's running in single threaded mode (see comment above) so it can't really spinlock.  The call above could though, in theory, but again that would be a symptom of a bigger problem I'd say.